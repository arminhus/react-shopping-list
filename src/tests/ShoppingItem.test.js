import React from "react";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import ShoppingItem from "../components/ShoppingItem";

Enzyme.configure({ adapter: new Adapter() });
describe("ShoppingItem Component Testing", () => {
  test("should render empty give no item", () => {
    const wrapper = shallow(<ShoppingItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render the item when supplied", () => {
    const testItem = { name: "test-item", checked: false };
    const wrapper = shallow(<ShoppingItem item={testItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should strike through checked items", () => {
    const testItem = { name: "check-item", checked: true };
    const wrapper = shallow(<ShoppingItem item={testItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
