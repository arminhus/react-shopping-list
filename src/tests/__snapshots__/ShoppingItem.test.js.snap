// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem Component Testing should render empty give no item 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  />
  <button
    type="button"
  >
    check
  </button>
  <button
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component Testing should render the item when supplied 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  >
    test-item
  </h2>
  <button
    type="button"
  >
    check
  </button>
  <button
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component Testing should strike through checked items 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": "line-through",
      }
    }
  >
    check-item
  </h2>
  <button
    type="button"
  >
    check
  </button>
  <button
    type="button"
  >
    delete
  </button>
</li>
`;
